src/features/amUI/users/useDebouncedFilteredRightHolders.ts

```diff
@@ -0,0 +1,76 @@
+import { getTotalNumOfPages, getArrayPage, debounce } from '@/resources/utils';
+import type { RightHolder } from '@/rtk/features/userInfo/userInfoApi';
+import { useGetRightHoldersQuery } from '@/rtk/features/userInfo/userInfoApi';
+import { useState, useMemo, useEffect } from 'react';
+
+const isSearchMatch = (searchString: string, rightHolder: RightHolder): boolean => {
+  const isNameMatch = rightHolder.name.toLowerCase().indexOf(searchString.toLowerCase()) > -1;
+  const isPersonIdMatch = rightHolder.personId === searchString;
+  const isOrgNrMatch = rightHolder.organizationNumber === searchString;
+  return isNameMatch || isPersonIdMatch || isOrgNrMatch;
+};
+
+export const useDebouncedFilteredRightHolders = (
+  searchString: string,
+  currentPage: number,
+  pageSize: number,
+) => {
+  const { data: rightHolders, isLoading: isFetching } = useGetRightHoldersQuery();
+  const [pageEntries, setPageEntries] = useState<RightHolder[]>([]);
+  const [numOfPages, setNumOfPages] = useState<number>(1);
+  const [searchResultLength, setSearchResultLength] = useState<number>(0);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const computePageEntries = () => {
+    if (!rightHolders) {
+      setPageEntries([]);
+      setNumOfPages(1);
+      setSearchResultLength(0);
+      setLoading(false);
+      return;
+    }
+
+    const filteredRightHolders: RightHolder[] = searchString ? [] : rightHolders;
```

Comment:
Since there is a possibility of filter being added later, maybe this should be called searchResults or something instead?

Reply:
👍🏼